### PR TITLE
Allow the volume of NVDA sounds to be set within NVDA.

### PR DESCRIPTION
**Pull request by jcsteh**
### Link to issue number:
Better solution for #1409. Addresses https://github.com/nvaccess/nvda/pull/14896#issuecomment-1594713278. Also addresses feedback I received on Mastodon regarding the NVDA sounds volume not appearing in the Windows Volume Mixer in some versions of Windows.

### Summary of the issue:
1. In #14697, I split NVDA speech and sounds into two separate Windows audio sessions to allow independent control of their levels. However, some users want these levels to be linked.
2. In #14896, I made it possible to have the sounds volume follow the voice volume. However, as per https://github.com/nvaccess/nvda/pull/14896#issuecomment-1594713278, some synths don't set the volume in a linear way, so the volumes don't match.
3. It was reported to me on Mastodon that in Windows 11 Insider builds, the "NVDA sounds" volume doesn't show up in the Windows Volume Mixer unless you actively play a sound while the Volume Mixer is running. This is not very intuitive, and if this is behaviour Microsoft intends to keep, it isn't going to work well for our users.
4. Having the volume of NVDA sounds controlled outside of NVDA is a little cumbersome.

### Description of user facing changes
1. There will now only be one entry for NVDA in the Windows Volume Mixer which will control all NVDA audio.
2. There will be a slider in NVDA Advanced Settings to control the volume of sounds.

### Description of development approach
1. Removed the ability to set a WASAPI session and volume.
2. Added a method to set the volume of a stream. This also allows you to set the volume of individual channels, though that's not used yet.
3. Added a slider in Advanced Settings to control the volume of NVDA sounds. This is disabled if the sound volume is following the voice volume.

### Testing strategy:
All the tests from #14896, plus:

1. Disabled "Volume of NVDA sounds follows voice volume".
2. Lowered the NVDA sounds volume to 10%.
3. Switched between focus and browse modes to verify that sounds played quieter.
4. Increased the NVDA sounds volume to 100%.
5. Switched between focus and browse modes to verify that sounds played louder.
6. Enabled "Volume of NVDA sounds follows voice volume".
7. Set the voice volume to 50%.
8. Switched between focus and browse modes to verify that sounds played quieter.

### Known issues with pull request:
No user facing issues. However:

1. I'm wondering whether I should make AudioPurpose, the purpose argument to WavePlayer and the setVolume method private (underscore prefixed) for now. I don't anticipate any further changes, but nor did I anticipate changes to the previous session API. Making it private would allow us to ensure this is stable internally before making it part of the public API. This might be overkill though.
2. This does change the WavePlayer API in beta (but not in release). I don't think that's considered API breaking if this gets merged to beta, but it's worth noting. It will be a problem if we don't merge this to beta.

### Change log entries:
In this entry in New features:
```
- You can now separately control the volume of NVDA sounds.
This can be done using the Windows Volume Mixer. (#1409)
```
The second line should be adjusted to say:
`This can be done in Advanced settings. (#1409)`

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by brunoprietog** [comment]
Thanks!

File: nvdaHelper/local/wasapi.cpp [hunk 1]
```diff
@@ -571,8 +552,10 @@ HRESULT wasPlay_resume(WasapiPlayer* player) {
 	return player->resume();
 }
 
-HRESULT wasPlay_setSessionVolume(WasapiPlayer* player, float level) {
-	return player->setSessionVolume(level);
+HRESULT wasPlay_setChannelVolume(WasapiPlayer* player, unsigned int channel,
+	float level
+) {
```
Comment (seanbudd) [inline nvdaHelper/local/wasapi.cpp:557]:
```suggestion
HRESULT wasPlay_setChannelVolume(
	WasapiPlayer* player,
	unsigned int channel,
	float level
) {
```

File: source/config/configSpec.py [hunk 2]
```diff
@@ -59,6 +59,7 @@
 	audioDuckingMode = integer(default=0)
 	wasapi = boolean(default=true)
 	soundVolumeFollowsVoice = boolean(default=false)
+	soundVolume = integer(default=100,min=0,max=100)
```
Comment (seanbudd) [inline source/config/configSpec.py:62]: Fix lint

```suggestion
	soundVolume = integer(default=100, min=0, max=100)
```

**Review by seanbudd — COMMENTED** [review]
Generally looks good to me

**Review by seanbudd — APPROVED** [review]
Thanks @jcsteh

**Comment by jcsteh** [comment]
> But what about other parameters of audio section? Are they meant to be profile dependant?

I don't see any reason they should be profile independent, though I'm not sure whether they all currently behave properly when you switch profiles. I can't see any reason the volume settings shouldn't switch properly across profiles.

> Also, is there any plan to move all those audio settings out of the advanced settings panel? I think that we should not wait for years before moving them in their own section.

I don't have a specific plan to do this, but yes, I agree they should probably be moved out of Advanced Settings once WASAPI is no longer a configurable option; i.e. once WASAPI is always enabled.

**Review by Qchristensen — APPROVED** [review]
Looks good.  This issue comes up from time to time, so it will make a number of users happy, thanks!

**Pull request merged**